src/main/java/org/sasanlabs/service/vulnerability/fileupload/UnrestrictedFileUpload.java

```diff
@@ -92,10 +96,10 @@ public UnrestrictedFileUpload() throws IOException, URISyntaxException {
                     "If you are running vulnerableApp as a Jar then UnrestrictedFileUpload will not work. "
                             + "For more information: https://github.com/SasanLabs/VulnerableApp/issues/255",
                     e);
-            if (root != null) {
+            if (root == null || !root.toFile().exists()) {
```

Comment:
will we be able to retrieve the files from temp location after this change?

Reply:
Good question. What I aimed for and tested is that my change does not impact the original behavior. E.g. I tested the app running as a jar, and also running with `./gradlew bootRun`. (I cannot use docker in my current setup.) In both cases the behavior stayed unchanged. However, I'm not 100% clear on how this is supposed to work. I don't see a way in the existing code to return files saved to a temp `root` location. When the `root` location stays a static location within the published web file structure, this gets served as all the other static files. However, when `root` gets modified to a tmp location, there is no way to load saved files from there. `PreflightController` does not have logic to serve files from a temp `root`.

I didn't enter a bug for this as I'm not 100% sure that I'm not misreading the code and this wasn't the focus of my work. But there might be an existing bug in there. My change should not impact this either way, the original logic should be unchanged.